lang/csharp/src/apache/main/Generic/PreresolvingDatumWriter.cs

```diff
@@ -232,6 +234,17 @@ private void WriteArray(WriteItem itemWriter, object array, Encoder encoder)
             encoder.WriteArrayEnd();
         }
 
+        /// <summary>
+        /// Serializes a logical value object by using the underlying logical type to convert the value
+        /// to its base value.
+        /// </summary>
+        /// <param name="schema">The logical schema.</param>
+        protected WriteItem ResolveLogical(LogicalSchema schema)
+        {
+            var baseWriter = ResolveWriter(schema.BaseSchema);
+            return (d, e) => baseWriter(schema.LogicalType.ConvertToBaseValue(d, schema), e);
```

Comment:
passing `schema` in as an argument to `ConvertToBaseValue` here seems really unorthodox, since it's a method on a property of the object itself. I have not bothered to dig deeper, but suspect this is indication of a factoring that isn't right.

Reply:
Some `LogicalType` implementations need to retrieve logical properties from the schema instance when creating a value of the underlying type. `Decimal` for example retrieves the `scale` property. The logical type is not the schema.

Reply:
ok, i see.